src/main/java/adris/altoclef/Settings.java

```diff
@@ -189,104 +176,75 @@
      */
-    private int[] importantItems = new int[] {
-            Item.getRawId(Items.ENCHANTED_GOLDEN_APPLE),
-            Item.getRawId(Items.ENDER_EYE),
+    private int[] importantItems = {
+            Item.getRawId(Items.ENCHANTED_GOLDEN_APPLE), Item.getRawId(Items.ENDER_EYE),
             // Don't throw away shulker boxes that would be pretty bad lol
-            Item.getRawId(Items.SHULKER_BOX),
-            Item.getRawId(Items.BLACK_SHULKER_BOX),
-            Item.getRawId(Items.BLUE_SHULKER_BOX),
-            Item.getRawId(Items.BROWN_SHULKER_BOX),
-            Item.getRawId(Items.CYAN_SHULKER_BOX),
-            Item.getRawId(Items.GRAY_SHULKER_BOX),
-            Item.getRawId(Items.GREEN_SHULKER_BOX),
-            Item.getRawId(Items.LIGHT_BLUE_SHULKER_BOX),
-            Item.getRawId(Items.LIGHT_GRAY_SHULKER_BOX),
-            Item.getRawId(Items.LIME_SHULKER_BOX),
-            Item.getRawId(Items.MAGENTA_SHULKER_BOX),
-            Item.getRawId(Items.ORANGE_SHULKER_BOX),
-            Item.getRawId(Items.PINK_SHULKER_BOX),
-            Item.getRawId(Items.PURPLE_SHULKER_BOX),
-            Item.getRawId(Items.RED_SHULKER_BOX),
-            Item.getRawId(Items.WHITE_SHULKER_BOX),
-            Item.getRawId(Items.YELLOW_SHULKER_BOX)
-    };
+            Item.getRawId(Items.SHULKER_BOX), Item.getRawId(Items.BLACK_SHULKER_BOX), Item.getRawId(Items.BLUE_SHULKER_BOX), Item.getRawId(
+            Items.BROWN_SHULKER_BOX), Item.getRawId(Items.CYAN_SHULKER_BOX), Item.getRawId(Items.GRAY_SHULKER_BOX), Item.getRawId(
+            Items.GREEN_SHULKER_BOX), Item.getRawId(Items.LIGHT_BLUE_SHULKER_BOX), Item.getRawId(Items.LIGHT_GRAY_SHULKER_BOX),
+            Item.getRawId(Items.LIME_SHULKER_BOX), Item.getRawId(Items.MAGENTA_SHULKER_BOX), Item.getRawId(Items.ORANGE_SHULKER_BOX),
+            Item.getRawId(Items.PINK_SHULKER_BOX), Item.getRawId(Items.PURPLE_SHULKER_BOX), Item.getRawId(Items.RED_SHULKER_BOX),
+            Item.getRawId(Items.WHITE_SHULKER_BOX), Item.getRawId(Items.YELLOW_SHULKER_BOX)
+    }; // TODO: 2021-05-22 please stop storing raw ids lol
+    // TODO: 2021-05-22 plus, might want to make this more generic so you don't have to specify it all. Again, configs
 
     /**
-     * These areas will not be mined.
-     * Used to prevent griefing
-     * or to define a "spawn protection" zone so
-     * the bot doesn't keep trying to break spawn protected
-     * blocks.
+     * These areas will not be mined. Used to prevent griefing or to define a "spawn protection" zone so the bot doesn't keep trying to
+     * break spawn protected blocks.
      */
-    private ProtectionRange[] areasToProtect = new ProtectionRange[] {
-
-    };
+    private ProtectionRange[] areasToProtect = { };
+    // TODO: 2021-05-22 bot should be able to detect when a server has "spawn protection", then use that to make some guesses about what
```

Comment:
Yeah, would be great. Haven't figured out how to do that yet though.

Reply:
basically, just do the following:
- break a block
- if it works, perfect.
- if the block magically comes back, break it again
- if it magically comes back *again*, then that means the server is preventing you from mining it. Assume something like a 4x4 chunk area around that has spawn protection. Move to the closest spot without spawn protection, and repeat. Do this until you find an area without spawn protection.

Reply:
Could work as an option, but I'm unsure if this would work in ALL scenarios. Will look into this.

"ProtectedAreas" can also be used for user bases and stuff like that, so will keep this.